src/ast/Node.hpp

```diff
@@ -165,17 +178,26 @@ bool Node::is(const T& test) const {
 	static_assert(
 	    std::is_same<T, math::Real>::value ||
 	    std::is_same<T, math::Rational>::value ||
+		std::is_same<T, math::ComplexReal>::value ||
+		std::is_same<T, math::ComplexRational>::value ||
 	    std::is_same<T, ast::FunctionCall>::value ||
 	    std::is_same<T, ast::String>::value ||
 	    std::is_same<T, ast::Identifier>::value, "invalid Node type" );
 	return is<T>() && get<T>() == test;
 }
 
+template<class T, class U, class... Ts>
+bool Node::is() const {
+	return Node::is<T>() || Node::is<U, Ts...>();
```

Comment:
This is linear. I think it can be done in constant time by using a smart visitor.